crates/sui-framework/sources/governance/sui_system.move

```diff
@@ -504,15 +507,17 @@ module sui::sui_system {
             wrapper.version = new_system_state_version;
             dynamic_field::add(&mut wrapper.id, wrapper.version, new_state);
         };
+        storage_rebate
     }
 
+    // TODO: keep rewards for safe mode epoch change too.
     /// An extremely simple version of advance_epoch.
     /// This is called in two situations:
     ///   - When the call to advance_epoch failed due to a bug, and we want to be able to keep the
     ///     system running and continue making epoch changes.
     ///   - When advancing to a new protocol version, we want to be able to change the protocol
     ///     version
-    public entry fun advance_epoch_safe_mode(
+    public(friend) fun advance_epoch_safe_mode(
```

Comment:
does this need to be friend? or can it be private now?

Reply:
It's friend for testing purpose but later on when we start keeping rewards for safe mode too we will need separate special `for_testing` functions so will make this one private at that time. Will add a TODO here.